ui/js/models/resultset.js

```diff
@@ -239,10 +239,18 @@ treeherder.factory('ThResultSetModel', ['$rootScope', '$http', '$location',
                     if (errorMsg === 'credentials must be given') {
                         errorMsg = 'Missing Taskcluster credentials! Please log out and back in again.';
                     }
-                    throw new Error(errorMsg);
+                    return $q.reject(new Error(errorMsg));
                 }
                 return $http.get(url).then(function (resp) {
                     let graph = resp.data;
+
+                    // Build a mapping of buildbot buildername to taskcluster tasklabel for bbb tasks
+                    let builderToTask = _.omit(_.invert(_.mapValues(graph, 'task.payload.buildername')), [undefined]);
+
+                    buildernames = _.map(buildernames, function(buildername) {
```

Comment:
It's a little confusing here to re-use the variable ``buildernames`` when these are now ``tasklabel``s.  (It was confusing to ME, anyway...  :)  )

I know it would require changing more lines down below, but perhaps it'd be clearer what you're working on to use a new var name?

Also, would you also be up for commenting what  lines 254 and 255 are doing and why?  It seems like 254 is unnecessary since ``graph`` is the source of the updated ``buildernames`` but maybe I'm missing something here.